### DISCUSSION TITLE
form with select field using ->relationship() fails when called from Action?

### DISCUSSION THREAD

**Discussion by wivaku** [filamentphp/filament#14847] [Help]
### Package

Actions

### Package Version

v3.2.124

### How can we help you?

I have Post and Note models (and Filament resources). 
And I have a Filament relation manager set up for the Post resource, to show a list of associated Notes.

The Post model uses a string key instead of a numeric ID.
The Note resource has a form, with fields for (Post) key and summary.
I reuse the form for both the Note resource and the Post resource (through ViewPost.php).

So: I can create a new Note (and select the associated Post), or I can select a Post and add a new Note (for that Post).

Everything works as expected when using a fixed list of options for the key Select field in the form.
But when using `->relationship('post', 'key')` in the form it works in the resource, but not when using it through action in `ViewPost.php`.

The error when pressing 'addNote' headerAction button:
```
Filament\Support\Services\RelationshipJoiner::prepareQueryForNoConstraints(): 
Argument #1 ($relationship) must be of type Illuminate\Database\Eloquent\Relations\Relation, 
null given, called in /.../vendor/filament/forms/src/Components/Select.php on line 765
```

The error is shown before the modal is shown.

Any suggestions how to make the Post relationship lookup work in both scenario's?

The code snippets:

```php
// PostResource/Pages/ViewPost.php

    protected function getHeaderActions(): array
    {
        return [
            // Actions\EditAction::make(),

            Action::make('addNote')
                ->fillForm(fn ($record): array => [
                    'key' => $record->key,
                ])
                ->form(fn($form) => NoteResource::form($form) )
                ->action(function (array $data, $record): void {
                    $record->notes()->create($data);
                })
        ];
    }
```

The form, which fails for option A (when called through `ViewPost.php`) and always works for option B.

```php
// NoteResource.php

    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\Select::make('key')
                    // option A: results in error when using Action in ViewPost
                    ->relationship(name:'post', titleAttribute:'key') 
                    // option B: works in both resource and Action in ViewPost
                    // ->options([
                    //  'key123' => 'key123'
                    // ])
                ,

                Forms\Components\TextInput::make('summary'),
            ]);
    }
```

**Comment by miiiwdy** [comment]
try using ->pluck()

**Reply by miiiwdy** [reply]
->options(Post::all()->pluck('key', 'id')),

**Reply by wivaku** [reply]
This would fetch all posts and that’s not feasible. I need it to be a database search.
